haystack/preview/dataclasses/answer.py

```diff
@@ -0,0 +1,26 @@
+from typing import Any, Dict, List, Optional
+from dataclasses import dataclass
+from haystack.preview.dataclasses.document import Document
+
+
+@dataclass
+class Answer:
+    data: Any
+    question: str
+    metadata: Dict[str, Any]
+
+
+@dataclass
+class ExtractiveAnswer(Answer):
+    data: Optional[str]
```

Comment:
I can't set this to a default value because of dataclass's implementation specifics. However, I am also not sure if a default value of `None` makes sense in this case.

Reply:
I already tried that and it doesn't work. The position of data is still before the other child fields because it is already defined in the parent. (That's also the inheritance issue I referred to above.)

Reply:
I think this is fine without a default :+1: